src/main/java/it/gov/pagopa/payment/instrument/service/idpaycode/PaymentInstrumentCodeServiceImpl.java

```diff
@@ -48,12 +49,18 @@ public GenerateCodeRespDTO generateCode(String userId, String initiativeId) {
     // generate clear code
     String clearCode = buildCode();
 
+    // generate Salt
+    String salt = generateSaltOrSecondFactor(16);
+
+    // generate secondFactor
+    String secondFactor = generateSaltOrSecondFactor(16);
+
     // encrypt clear code
-    String idpayCode = encryptCodeService.encryptIdpayCode(clearCode);
+    String idpayCode = encryptCodeService.encryptIdpayCode(clearCode, secondFactor, salt);
```

Comment:
let's give more meaningfull names
```suggestion
    String hashedPinBlock= encryptCodeService.encryptIdpayCode(clearCode, secondFactor, salt);
```